src/main/java/org/wildfly/security/auth/server/MechanismConfiguration.java

```diff
@@ -41,14 +43,19 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class MechanismConfiguration {
+
+    private final MechanismConfiguration parent;
     private final NameRewriter preRealmRewriter;
     private final NameRewriter postRealmRewriter;
     private final NameRewriter finalRewriter;
     private final RealmMapper realmMapper;
     private final Map<String, MechanismRealmConfiguration> mechanismRealms;
     private final List<SecurityFactory<Credential>> serverCredentialFactories;
+    private final Map<String, MechanismConfiguration> serverSpecificConfiguration;
```

Comment:
I will concede and switch to host ;-)  Just checked FreeIPA which is a project this will most likely be paired with and I see they also use 'host'.